corehq/ex-submodules/fluff/signals.py

```diff
@@ -136,6 +137,53 @@ def get_tables_to_rebuild(diffs, table_names):
     }
 
 
+def get_tables_with_index_changes(diffs, table_names):
+    return {
+        diff.table_name
+        for diff in diffs
+        if diff.table_name in table_names and diff.type in DiffTypes.INDEX_TYPES
+    }
+
+
+def _get_indexes_to_change(raw_diffs, table_names):
+    # raw diffs come in as a list of (action, index)
+    indexes = [diff for diff in raw_diffs if diff[0] in DiffTypes.INDEX_TYPES]
+    indexes_by_table_and_col = defaultdict(list)
+
+    for action, index in indexes:
+        if index.table.name not in table_names:
+            continue
+
+        column_names = tuple(index.columns.keys())
+        indexes_by_table_and_col[(index.table.name, column_names)].append((action, index))
+
+    indexes_to_change = []
+
+    for indexes in indexes_by_table_and_col.values():
+        if len(indexes) == 1:
+            indexes_to_change.append(indexes[0])
+        else:
+            # tests when alembic tries to remove/add the same index
+            assert len(indexes) == 2
```

Comment:
> how does this happen?

I'm not sure why this is but when alembic generates an index it can generate a list like:

[(add_index, index_with_name_a_on_col),(remove_index, index_with_name_b_on_col)]

But for som reason when if you add the index with name a, the name b will get added consistently. It also makes sure we don't remove/add any indexes that were added manually on icds

> could it ever be more than 2

I don't think so, but that's why I added the assertions. I think since this is based on table and column that the only thing that could happen would be 1 add and one remove. I'd rather something like this fail hard before altering the database, but happy to look into any other solutions you might have